xarray/core/formatting.py

```diff
@@ -114,9 +114,11 @@ def calc_max_rows_last(max_rows: int) -> int:
 
 def format_timestamp(t):
     """Cast given object to a Timestamp and return a nicely formatted string"""
-    # Timestamp is only valid for 1678 to 2262
     try:
-        datetime_str = str(pd.Timestamp(t))
+        timestamp = pd.Timestamp(t)
+        datetime_str = timestamp.isoformat()
+        if datetime_str != "NaT":
+            datetime_str = datetime_str.replace("T", " ")
```

Comment:
Oops sorry for not remembering to override the default separator.  Was there a reason you did not go with something like this instead?  It seems to do the right thing for `NaT`:
```suggestion
        datetime_str = timestamp.isoformat(sep=" ")
```

Reply:
no, there was not. I just forgot that that was an option.